cg/meta/workflow/rnafusion.py

```diff
@@ -0,0 +1,299 @@
+import pandas as pd
+import os
+
+import yaml
+from pydantic import ValidationError
+from cg.constants import DataDelivery, Pipeline
+from cg.exc import RnafusionStartError, CgError
+from cg.meta.workflow.analysis import AnalysisAPI
+from cg.meta.workflow.fastq import RnafusionFastqHandler
+from cg.models.cg_config import CGConfig
+from cg.store import models
+from cg.utils import Process
+from cg import resources
+from datetime import datetime
+from subprocess import CalledProcessError
+
+
+LOG = logging.getLogger(__name__)
+
+
+class RnafusionAnalysisAPI(AnalysisAPI):
+    """Handles communication between RNAFUSION processes
+    and the rest of CG infrastructure"""
+
+    def __init__(
+        self,
+        config: CGConfig,
+        pipeline: Pipeline = Pipeline.RNAFUSION,
+    ):
+        super().__init__(config=config, pipeline=pipeline)
+        self.root_dir = config.rnafusion.root
+        self.nfcore_pipeline_path = config.rnafusion.pipeline_path
+        self.references = config.rnafusion.references
+        self.profile = config.rnafusion.profile
+        self.conda_env = config.rnafusion.conda_env
+
+    @property
+    def root(self) -> str:
+        return self.root_dir
+
+    @property
+    def fastq_handler(self):
+        return RnafusionFastqHandler
+
+    @property
+    def process(self):
+        if not self._process:
+            self._process = Process(self.config.rnafusion.binary_path, None, None, self.conda_env)
+        return self._process
+
+    def get_case_path(self, case_id: str) -> Path:
+        """Returns a path where the rnafusion case should be located"""
+        return Path(self.root_dir, case_id)
+
+    def get_case_config_path(self, case_id: str) -> Path:
+        """Generates a path where the Rnafusion samplesheet for the case_id should be located."""
+        return Path((self.get_case_path(case_id)), case_id + "_samplesheet.csv")
+
+    def make_case_folder(self, case_id: str) -> None:
+        """Make the case folder where rnafusion analysis should be located"""
+        self.get_case_path(case_id)
+        os.makedirs(self.get_case_path(case_id), exist_ok=True)
+
+    def write_samplesheet(self, case_id: str, strandedness="reverse") -> None:
+        """Write samplesheet for rnafusion analysis in case folder"""
+        case_obj = self.status_db.family(case_id)
+        for link in case_obj.links:
+            file_collection = pd.DataFrame(self.gather_file_metadata_for_sample(link.sample))
+            file_collection = file_collection.sort_values(by=["path"])
+            fastq_r1 = file_collection[file_collection["read"] == 1]["path"].to_list()
+            fastq_r2 = file_collection[file_collection["read"] == 2]["path"].to_list()
+            samplesheet = pd.DataFrame(
+                list(zip(fastq_r1, fastq_r2)), columns=["fastq_1", "fastq_2"]
+            )
+            samplesheet["sample"] = case_id
+            samplesheet["strandedness"] = strandedness
```

Comment:
Difficult if I understand you correctly, because values are set differently for the different keys. Strandedness needs to just be repeated for each read lanes (so there is interdependence between the keys/values pairs) but this will be different for raredisease as the trio/phenotype setup is not following the same logic. Or do you mean a smaller scope of a function for just filling the dictionary?